### DISCUSSION TITLE
Any help for how to separate TipTap content into pages?

### DISCUSSION THREAD

**Discussion by ancientstraits** [ueberdosis/tiptap#4806] [General Questions]
So for a project I have to do, the client has asked me to separate the content into pages, like in Google Docs. However, information on how to do this is few and far between. I found [this project](https://github.com/todorstoev/prosemirror-pagination), but I don't know if it works, because it is 4 years old, and the code is extremely confusing. Then, I had an idea to make a version of the HorizontalRule that is actually a div that makes it look like there is a break in the page between the two pages. Then, I could add these `PageBreak` nodes at specific points in the editor to separate the content into pages. However, I have no idea how to do this. I want to make an extension command to do this, but I would have to know the size of everything, once it is rendered into HTML. It would be a bit easier if I could use `editor.getHTML()`, but that function does not work in custom commands, since you cannot do `const { editor } = useCurrentEditor()` in editor commands. So does anyone have any pointers on what I could do?

I already checked [this discussion](https://github.com/ueberdosis/tiptap/discussions/2249), but it was not that useful. I really *need* to add pagination to the editor, since the client requested it. Also, I do not think `pagedjs` can paginate content. I really don't know, since its website is extremely vague, and its examples are just books.

**Comment by Cassielxd** [comment]
Hello, my custom implementation may be helpful to you
[](https://github.com/Cassielxd/CassieEditor)

**Reply by Cassielxd** [reply]
[https://github.com/Cassielxd/CassieEditor](url)

**Reply by patelneeraj** [reply]
Hey, your repository is no more, can you please explain me how did you to do it?

**Reply by Cassielxd** [reply]
https://github.com/Cassielxd/CassieEditor
